### PR TITLE
Iso 8601 date format for the broadcast message

### PR DESCRIPTION
**Pull request by sudosimi**

### CODE REVIEW

File: src/main/groovy/io/sdkman/service/TextService.groovy [hunk 1]
```diff
@@ -34,19 +35,22 @@ class TextService {
...
-    private static buildMessage(broadcasts) {
-        def output = ""
-        broadcasts.each { broadcast ->
-            output += "* ${getDateInstance(SHORT, Locale.UK).format(broadcast.date)}: $broadcast.text\n"
-        }
-        output
+    private buildMessage(broadcasts) {
+        broadcasts
+            .collect{"* ${toISO8601(it.date)}: $it.text" + lineSeparator()}
+            .join()
```
Comment (marc0der) [inline src/main/groovy/io/sdkman/service/TextService.groovy]: You can simply do line separation as a parameter to the join function:
```
.join(lineSeparator())
```

File: src/test/groovy/io/sdkman/service/TextServiceSpec.groovy [hunk 2]
```diff
@@ -27,6 +28,13 @@ class TextServiceSpec extends Specification {
     Broadcast broadcast1 = new Broadcast(id: 1, text: "text1", date: new Date())
     Broadcast broadcast2 = new Broadcast(id: 2, text: "text2", date: new Date())
 
+
+    Date iso8601Date = new Date().copyWith(
+            year: 2019,
+            month: Calendar.FEBRUARY,
+            dayOfMonth: 11)
+    Broadcast iso8601Broadcast = new Broadcast(id: "3",text: "Groovy 2.7.1 released!", date: iso8601Date)
+
```
Comment (marc0der) [inline src/test/groovy/io/sdkman/service/TextServiceSpec.groovy:32]: I would move this into the test body itself so everything is close together. Spock allows for a `given:` block where this code would fit well.

**Review by marc0der — COMMENTED** [review]
Looks great, well done! Just one or two small things that I would address. Also we need to look into why that build is failing.

File: src/main/groovy/io/sdkman/service/TextService.groovy (hunk 1 above)
Comment (sudosimi) [reply src/main/groovy/io/sdkman/service/TextService.groovy]: Unfortunately I'd have to add a + lineSeparator(). The Cuke expects a `lineSeparator()` at the end and the `join()` only adds it between the elements and not at the end. 
The code would look bulkier like: `.join(lineSeparator()) + lineSeparator()` in order for the cukes to pass.

File: src/test/groovy/io/sdkman/service/TextServiceSpec.groovy (hunk 2 above)
Comment (sudosimi) [reply src/test/groovy/io/sdkman/service/TextServiceSpec.groovy:32]: Ok, will change. That what I've done initially, but I moved it up to keep it consistent with the other two broadcasts.

File: src/main/groovy/io/sdkman/service/TextService.groovy (hunk 1 above)
Comment (marc0der) [reply src/main/groovy/io/sdkman/service/TextService.groovy]: I'd prefer it that way tbh.

**Comment by marc0der** [comment]
Great work, and thanks for the contribution!

**Pull request merged**